### ISSUE TITLE
1.1.407 breaks the use of pydantic.dataclasses attributes

### ISSUE THREAD

**Issue by engnatha** [microsoft/pyright#11060]
**Describe the bug**
When using the `@pydantic.dataclasses.dataclass` decorator, any reference to a class member (attribute, method, etc) is raised with a `reportedAttributeAccessIssue`.

Notably, the builtin `dataclasses` library does not have this issue.

**Code or Screenshots**
Here is some minimal code that reproduces this failure.

```python
import pydantic


@pydantic.dataclasses.dataclass(frozen=True)
class SubModel:
    a: int
    b: int


@pydantic.dataclasses.dataclass(frozen=True)
class Model:
    foo: SubModel
    bar: int

    @property
    def a_from_foo(self) -> int:
        return self.foo.a


if __name__ == '__main__':
    m = Model(**{'foo': {'a': 1, 'b': 2}, 'bar': 3})
    print('model', m)
    print('a_from_foo', m.a_from_foo)
```

The `pydantic` version used in testing is `2.9.2`

The script is functional.
```
python pydantic_demo.py 
model Model(foo=SubModel(a=1, b=2), bar=3)
a_from_foo 1
```
But pyright 1.1.407 flags this as an error.
```
pnpm exec pyright --version
pyright 1.1.407
pnpm exec pyright pydantic_demo.py 
pydantic_demo.py
  pydantic_demo.py:17:25 - error: Cannot access attribute "a" for class "PydanticDataclass"
    Attribute "a" is unknown (reportAttributeAccessIssue)
  pydantic_demo.py:23:27 - error: Cannot access attribute "a_from_foo" for class "PydanticDataclass"
    Attribute "a_from_foo" is unknown (reportAttributeAccessIssue)
2 errors, 0 warnings, 0 informations
```
pyright 1.1.406 is happy with the file
```
pnpm exec pyright --version
pyright 1.1.406
pnpm exec pyright pydantic_demo.py 
0 errors, 0 warnings, 0 informations
```

**Comment by engnatha** [comment]
Haven't explored too much yet, but I did notice the release notes call out #11018. Seems suspiciously correlated to this breaking, but it could be a red herring.

**Comment by erictraut** [comment]
Thanks for the bug report. I've bisected the regression to #11018, which confirms your theory. I'll need to investigate further.

**Comment by gvanrossum-ms** [comment]
I got hit hard with this today.

**Comment by helmut-hoffer-von-ankershoffen** [comment]
Do we have an update here? Would like to remove 

"pyright>=1.1.406,<1.1.407", # Regression in 1.1.407, see https://github.com/microsoft/pyright/issues/11060 

from multiple code bases ;)

**Comment by erictraut** [comment]
@helmut-hoffer-von-ankershoffen, this will be fixed in the next release. I don't have a specific ETA yet. It will probably be at least a couple more weeks.

**Comment by helmut-hoffer-von-ankershoffen** [comment]
Ty @erictraut

**Comment by matheuskknd** [comment]
Any updates about the issue? Visual Studio Code lastest versions seem to be using this buggy version of pyright now..

**Comment by engnatha** [comment]
You can select the version you want vscode to use in the settings. I have `"python.analysis.pyrightVersion": "1.1.406",` in mine to get around this.

**Comment by matheuskknd** [comment]
It doesn't seem to fully fix the issue on vscode:

<img width="1457" height="859" alt="Image" src="https://github.com/user-attachments/assets/22117777-ef3a-485a-a318-98ab40bdff00" />

<img width="803" height="281" alt="Image" src="https://github.com/user-attachments/assets/c1323b1c-0f9b-44da-81aa-33dc3de00cd0" />

<img width="836" height="288" alt="Image" src="https://github.com/user-attachments/assets/60201195-2363-4968-acb7-4eb1ce6ccadf" />

EDIT: using same pyright==1.1.401 via CLI show no error at all.

<img width="708" height="78" alt="Image" src="https://github.com/user-attachments/assets/bc3e3fc2-f513-4c15-8b40-ac6911445780" />

**Comment by rchiodo** [comment]
Using this setting: "python.analysis.pyrightVersion": "1.1.406", only applies to the diagnostics. It doesn't change how hover works.

**Comment by matheuskknd** [comment]
Thank you!

**Comment by m4reko** [comment]
I just pinned my pylance version to 2025.9.1 from the VS Code extension page which seems to be working fine.

<img width="481" height="175" alt="Image" src="https://github.com/user-attachments/assets/72645f8c-928e-48d1-85f9-d4b2b559020e" />

**Comment by rchiodo** [comment]
This is addressed in pyright 1.1.408 and will ship in a future version of Pylance.
